src/language/tsLanguageService.ts

```diff
@@ -0,0 +1,215 @@
+import { Connection } from 'vscode-languageserver/node';
+import type {
+  SignatureHelp,
+  SignatureInformation,
+  ParameterInformation,
+} from 'vscode-languageserver/node';
+import ts from 'typescript';
+import { TextDocument, Position } from 'vscode-languageserver-textdocument';
+import { readFileSync } from 'fs';
+import { join } from 'path';
+
+type TypeScriptServiceHost = {
+  getLanguageService(jsDocument: TextDocument): ts.LanguageService;
+  getCompilationSettings(): ts.CompilerOptions;
+  dispose(): void;
+};
+
+const contents: { [name: string]: string } = {};
```

Comment:
Should this be global or should it be a property on the `TypeScriptService` instance?

(also: you’ll probably want to use a Map here or `Object.create(null)` instead of `{}`, otherwise weird things happen if something accesses e.g. `contents['__proto__']`)